### PR TITLE
Correctly handle no game history data existing.

### LINKED ISSUE #7489: MP Lobby: Assertion error in Match History tabs
### Describe the desired feature

I understand Match History was only implemented in 1.17 so this isn't applicable for 1.16. Since I don't play MP, I assume this is why have no history. However, attempting to use the history function results in an unresponsive UI while the client attempts to find data that isn't there. This is confirmed by log messages:

> error network: Timed out waiting for history data, returning nothing

Time-out is supposed to be only 5 seconds, but it still feels strange to have the UI do nothing (and block redraws). Perhaps attempt the network transfer asynchronously - maybe 5 seconds is too short an interval to bother with showing a progress bar or any such feedback.

Additionally, the history dialogue should probably not be shown at all if no history data is found - display some text to the user instead, something simple like 'no history found'.

### PR DESCRIPTION
**Pull request by Pentarctagon**
Fixes #7489

---

@Wedge009 I'm not able to reproduce the crash you mentioned in that issue. Could you get a stack trace for it?

### CODE REVIEW

**Comment by Pentarctagon** [comment]
Alright, the crash has been fixed as well now. Barring anything else being said, I plan to merge this in the next day or two.

**Comment by Wedge009** [comment]
Thanks, sounds good to me.

File: src/server/common/resultsets/game_history.cpp [hunk 1]
```diff
@@ -52,6 +52,13 @@ void game_history::read(mariadb::result_set_ref rslt)
 std::unique_ptr<simple_wml::document> game_history::to_doc()
 {
     auto doc = std::make_unique<simple_wml::document>();
+
+    if(results.size() == 0)
+    {
+        doc->set_attr_dup("no_data", "yes");
```
Comment (soliton-) [inline src/server/common/resultsets/game_history.cpp:58]: Why is this necessary? Just no `[game_history_result]` means the same thing, no? If more structure is needed then the existing WML should be wrapped in `[game_history_results]` or so. (Certainly the way just an error attribute is sent right now is not so great.)

Comment (Pentarctagon) [reply src/server/common/resultsets/game_history.cpp:58]: If an unrelated response is sent from the server to the client, then it would also have no [game_history_result].

Comment (soliton-) [reply src/server/common/resultsets/game_history.cpp:58]: If an unrelated response contains a `no_data` attribute then you have a similar issue. Thus the suggestion to always send a tag like `[game_history_results]`.

Comment (Pentarctagon) [reply src/server/common/resultsets/game_history.cpp:58]: I will update this then once the PR for storing a leader's information in the database has been resolved.
